graphql-jpa-query-schema/src/main/java/com/introproventures/graphql/jpa/query/schema/impl/QraphQLJpaBaseDataFetcher.java

```diff
@@ -132,15 +126,15 @@ public QraphQLJpaBaseDataFetcher(EntityManager entityManager,
 
     @Override
     public Object get(DataFetchingEnvironment environment) {
-        return getQuery(environment, environment.getFields().iterator().next(), true).getResultList();
+        return getQuery(environment, environment.getField(), true).getResultList();
     }
 
     protected TypedQuery<?> getQuery(DataFetchingEnvironment environment, Field field, boolean isDistinct) {
         CriteriaBuilder cb = entityManager.getCriteriaBuilder();
         CriteriaQuery<?> query = cb.createQuery((Class<?>) entityType.getJavaType());
         Root<?> from = query.from(entityType);
 
-        DataFetchingEnvironment queryEnvironment = DataFetchingEnvironmentBuilder.newDataFetchingEnvironment(environment)
+        DataFetchingEnvironment queryEnvironment = DataFetchingEnvironmentImpl.newDataFetchingEnvironment(environment)
```

Comment:
@negesti I think it is better to extract `DataFetchingEnvironmentImpl.newDataFetchingEnvironment` into protected class method `newDataFetchingEnvironment`.